subsys/bluetooth/host/scan.c

```diff
@@ -516,6 +528,8 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 	while (num_reports--) {
 		struct bt_hci_evt_le_ext_advertising_info *evt;
 		struct bt_le_scan_recv_info adv_info;
+		uint8_t size_to_copy;
```

Comment:
```suggestion
		size_t size_to_copy;
```

Reply:
I used `uint8_t` as the size to be copied is always in the range of an uint8_t because of the length field of the report.
I'm fine with changing it to `size_t`, but I see no reason for doing it.

Reply:
A couple of reasons:
1) Semantically `size_t` is used to denote sizes (which this clearly is). 
2) At one point you do 
```
size_to_copy = net_buf_max_len(ext_scan_buf) - evt->length;
```
and net_buf_max_len returns a uint16_t; even with the decrement the remaining value (e.g. 1650 - 200) may overflow, so at the very least it should be a uint16_t
3) It is used as an argument to `net_buf_add_mem` which also takes a `size_t`

Reply:
- [ ] Use `size_t`